### ISSUE TITLE
Print spack concretization output on stdout and a separate log file

### ISSUE THREAD

**Issue by rfhaque** [llnl/benchpark#159]
Currently `benchpark setup` prints the spack concretization details in `<workspace-dir>/workspace/logs/setup.<time>.out` file. In case of multiple experiments, this concretization log appears only once per each set of experiments built using the same config options. This makes it difficult to track compilation output for a given experiment.  
If will be highly desirable to have the spack concretization output printed on the stdout and its own separate log file. Additionally, it will also be nice to have a mode like `benchpark setup --dry-run` which simply prints the concretization log for a given set of experiments.

**Comment by pearce8** [comment]
@joshuamvining @michaelmckinsey1 Another utility request you could solve while working library versions from spack concretization

**Comment by scheibelp** [comment]
FWIW you can get Spack to print out the results of the concretization after it is done:

```
(typical experiment/system init + setup)
./bin/benchpark setup system/experiment wkp/
. wkp/setup.sh
cd ./wkp/system/experiment/workspace/
ramble --disable-logger --workspace-dir . workspace setup

(then to look at the concretization result)
find . -name "spack.yaml"
(take spack.yaml off the resulting path path)
spack -e <path-to-env-dir>/ spec
```

**Comment by rfhaque** [comment]
> FWIW you can get Spack to print out the results of the concretization after it is done:
> 
> ```
> (typical experiment/system init + setup)
> ./bin/benchpark setup system/experiment wkp/
> . wkp/setup.sh
> cd ./wkp/system/experiment/workspace/
> ramble --disable-logger --workspace-dir . workspace setup
> 
> (then to look at the concretization result)
> find . -name "spack.yaml"
> (take spack.yaml off the resulting path path)
> spack -e <path-to-env-dir>/ spec
> ```

This works as expected
